list/list_test.go

```diff
@@ -63,105 +84,192 @@ func (suite *TestSuite) TestFunctionality() {
+func (suite *ListTestSuite) TestListNoConfig() {
+	err := List([]string{"list"}, "")
+	assert.EqualError(suite.T(), err, "failed to load config file, reason: failed to read the configuration file")
+}
 
-	rescueStdout = os.Stdout
+func (suite *ListTestSuite) TestListFiles() {
+	rescueStdout := os.Stdout
 	r, w, _ := os.Pipe()
 	os.Stdout = w
 
 	rescueStderr := os.Stderr
 	errR, errW, _ := os.Pipe()
 	os.Stderr = errW
 
-	os.Args = []string{"list"}
-	err = List(os.Args, configPath.Name())
+	err := List([]string{"list"}, suite.configPath)
 	assert.NoError(suite.T(), err)
 
 	_ = w.Close()
 	os.Stdout = rescueStdout
 	listOutput, _ := io.ReadAll(r)
-	msg1 := fmt.Sprintf("%v", filepath.Base(testfile.Name()))
-	assert.Contains(suite.T(), string(listOutput), msg1)
+	_ = r.Close()
+	assert.Contains(suite.T(), string(listOutput), fmt.Sprintf("%v", filepath.Base(suite.testFilePath)))
+
+	// Check that host_base is in the error output, not in the stdout
+	expectedHostBase := fmt.Sprintf("Remote server (host_base): %s", suite.s3HTTPServer.URL)
+	assert.NotContains(suite.T(), string(listOutput), expectedHostBase)
 
 	_ = errW.Close()
 	os.Stderr = rescueStderr
 	listError, _ := io.ReadAll(errR)
+	_ = errR.Close()
 
-	// Check that host_base is in the error output, not in the stdout
-	expectedHostBase := "Remote server (host_base): " + ts.URL
-	assert.NotContains(suite.T(), string(listOutput), expectedHostBase)
 	assert.Contains(suite.T(), string(listError), expectedHostBase)
 }
+
+func (suite *ListTestSuite) TestListDatasets() {
+	rescueStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	err := List([]string{"list", "-url", suite.downloadMockHTTPServer.URL, "-datasets"}, suite.configPath)
+	assert.NoError(suite.T(), err)
+
+	_ = w.Close()
+	os.Stdout = rescueStdout
+	listOutput, _ := io.ReadAll(r)
+	_ = r.Close()
+	assert.Contains(suite.T(), string(listOutput), fmt.Sprintf("%v", "TES01 \t 3 \t 3.1 kB"))
+}
+
+func (suite *ListTestSuite) TestListDataset() {
+	rescueStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	err := List([]string{"list", "-url", suite.downloadMockHTTPServer.URL, "-dataset", "TES01"}, suite.configPath)
+	assert.NoError(suite.T(), err)
+
+	_ = w.Close()
+	os.Stdout = rescueStdout
+	listOutput, _ := io.ReadAll(r)
+	_ = r.Close()
+	assert.Contains(suite.T(), string(listOutput), fmt.Sprintf("%v", "FileID               \t Size       \t Path\nfile1id \t 1.0 kB \t files/file1.c4gh\nfile2id \t 1.0 kB \t files/file2.c4gh\ndummyFile \t 1.0 kB \t files/dummy-file.txt.c4gh\nDataset size: 3.1 kB"))
+}
+
+func (suite *ListTestSuite) TestListDatasetNoUrl() {
+	err := List([]string{"list", "-dataset", "TES01"}, suite.configPath)
+	assert.EqualError(suite.T(), err, "invalid base URL")
+}
+func (suite *ListTestSuite) TestListDatasetsNoUrl() {
+	err := List([]string{"list", "-dataset", "TES01"}, suite.configPath)
+	assert.EqualError(suite.T(), err, "invalid base URL")
+}
+
+func (suite *ListTestSuite) generateDummyToken() string {
```

Comment:
Since accessToken has been used (and hardcoded) in other modules, e.g. upload, it is better to add this function in `helpers.go`.